app/ide-desktop/lib/dashboard/src/services/ProjectManager.ts

```diff
@@ -445,6 +445,13 @@ export default class ProjectManager {
       path == null ? this.rootDirectory : getDirectoryAndName(path).directoryPath
     const fullParams: RenameProjectParams = { ...params, projectsDirectory: directoryPath }
     await this.sendRequest('project/rename', fullParams)
+    const state = this.internalProjects.get(params.projectId)
+    if (state?.state === backend.ProjectState.opened) {
+      this.internalProjects.set(params.projectId, {
```

Comment:
Maybe it'd be better to reload that state instead of using direct mutation?

Reply:
not possible, even if it made sense (i don't think it does; we know the exact final state). the info we need is only available as a return value from `openProject`, and we cannot `openProject` a project that is already open